src/main/scala/io/iohk/ethereum/blockchain/sync/regular/FetcherService.scala

```diff
@@ -0,0 +1,86 @@
+package io.iohk.ethereum.blockchain.sync.regular
+
+import akka.util.ByteString
+import cats.data.EitherT
+import io.iohk.ethereum.domain.{Block, BlockBody, BlockHeader, BlockchainReader}
+import io.iohk.ethereum.network.Peer
+import io.iohk.ethereum.network.p2p.messages.ETH62.{BlockBodies, BlockHeaders}
+import monix.eval.Task
+import io.iohk.ethereum.blockchain.sync.PeerRequestHandler.RequestFailed
+import io.iohk.ethereum.consensus.validators.BlockValidator
+import io.iohk.ethereum.utils.Config.SyncConfig
+import cats.implicits._
+
+import scala.annotation.tailrec
+
+//not used atm, a part of the future ExecutionSync
+class FetcherService(validator: BlockValidator, blockchainReader: BlockchainReader, syncConfig: SyncConfig) {
+  val batchSize = syncConfig.blockHeadersPerRequest
```

Comment:
I think that this is an unfortunate pattern. the class does not depend on the whole sysconfig class - it depends on one field from that class. how can we better express such dependencies?
if it's one field perhaps a tagged number would be good enough?

Reply:
I'm not sure I know what's a tagged number🤔